### PR TITLE
libspdm x509 parts

### PR DESCRIPTION
**Pull request by julek-wolfssl**
- RsaFunctionPrivate: detect when only n,e,d are available
- wolfSSL_EVP_add_digest: return success
- wolfSSL_EVP_add_cipher: return success
- wolfSSL_BN_bin2bn: accept NULL data if len is 0 (checked in mp_read_unsigned_bin)
- wolfssl_read_bio: advance correct bio
- wolfSSL_X509_set_ext: return raw extension data for BASIC_CA_OID
- Implement
  - sk_X509_EXTENSION_free
  - d2i_EC_PUBKEY_bio
  - d2i_RSA_PUBKEY_bio
  - d2i_X509_REQ_INFO
  - X509_REQ_INFO_free
  - ASN1_TIME_set_string_X509

### CODE REVIEW

**Comment by julek-wolfssl** [comment]
Failures (not sure about `fips-ready-config`) not related to PR.

**Comment by douzzer** [comment]
retest this please

**Review by douzzer — CHANGES_REQUESTED** [review]
found some minor stuff.

File: wolfcrypt/test/test.c [hunk 1]
```diff
@@ -25546,7 +25546,7 @@ WOLFSSL_TEST_SUBROUTINE wc_test_ret_t openssl_test(void)
             return WC_TEST_RET_ENC_NC;
         }
 
-        if (EVP_CIPHER_CTX_block_size(NULL) != WC_NO_ERR_TRACE(BAD_FUNC_ARG))
+        if (EVP_CIPHER_CTX_block_size(NULL) != WOLFSSL_FAILURE)
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply wolfcrypt/test/test.c:25549]: Fixed

File: wolfcrypt/test/test.c [hunk 2]
```diff
@@ -25557,7 +25557,7 @@ WOLFSSL_TEST_SUBROUTINE wc_test_ret_t openssl_test(void)
         if (EVP_CIPHER_CTX_block_size(en) != en->block_size)
             return WC_TEST_RET_ENC_NC;
 
-        if (EVP_CIPHER_block_size(NULL) != WC_NO_ERR_TRACE(BAD_FUNC_ARG))
+        if (EVP_CIPHER_block_size(NULL) != WOLFSSL_FAILURE)
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply wolfcrypt/test/test.c:25560]: Fixed

File: wolfcrypt/test/test.c [hunk 3]
```diff
@@ -25574,11 +25574,8 @@ WOLFSSL_TEST_SUBROUTINE wc_test_ret_t openssl_test(void)
         if (en->flags != 42)
             return WC_TEST_RET_ENC_NC;
 
-        if (EVP_CIPHER_CTX_set_padding(NULL, 0) !=
-            WC_NO_ERR_TRACE(BAD_FUNC_ARG))
-        {
+        if (EVP_CIPHER_CTX_set_padding(NULL, 0) != WOLFSSL_FAILURE)
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply wolfcrypt/test/test.c:25577]: Fixed

**Review by douzzer — APPROVED** [review]

**Pull request merged**